### PR TITLE
[TensorExpr] Remove 'Placeholder' class.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64887
* #64886

BufHandle has exactly the same functionality and should be used instead.

Differential Revision: [D30889483](https://our.internmc.facebook.com/intern/diff/D30889483)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30889483).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30889483).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30889483).

**Review by navahgar — APPROVED** [review]
LGTM

File: torch/csrc/jit/tensorexpr/reduction.h [hunk 1]
```diff
@@ -26,9 +26,6 @@ class TORCH_API Reducer {
   Reducer(ExprHandle init, ReduceInteraction& interaction)
       : init_(init.node()), interaction_(interaction) {}
 
-  Reducer(ExprHandle init, ReduceInteraction& interaction, Placeholder& buf)
```
[the remark this answers is not in the document]
Comment (ZolotukhinM) [reply torch/csrc/jit/tensorexpr/reduction.h:29]: Unclear :)

**Pull request closed without merging**